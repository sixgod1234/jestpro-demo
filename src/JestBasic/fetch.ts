import axios from 'axios';

export default {
 // mock
  async fetchPostsList(callback) {
    return axios.get('http://localhost:3005/testData').then(res => {
      callback(res.data);
    })
  },

//   异步测试
// 1.普通返回
   fetchData(callback) {
     axios.get('http://localhost:3005/testData').then(res => {
        callback(res.data);
    })
 },
   //   2.直接返回promise
    fetchData2() {
        return axios.get('http://localhost:3005/testData')
    },

  //   2.错误接口
    fetchData3() {
        return axios.get('http://localhost:3005/testDatakkk')
    },
}