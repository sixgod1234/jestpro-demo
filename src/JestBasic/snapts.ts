export const generateConfig = () => {
    return {
      server: 'http://localhost',
      port: 8080,
      domain: 'localhost',
      time:"2019",
      age:20
    }
  }
export const generateAnotherConfig = () => {
    return {
      server: 'http://localhost',
      port: 8080,
      domain: 'localhost',
      time:new Date(),
      age:20
    }
  }
