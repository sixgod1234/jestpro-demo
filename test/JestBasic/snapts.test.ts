import { generateConfig } from "../../src/JestBasic/snapts";

test("测试 generateConfig ", () => {
  // 1.普通匹配，每次修改配置时，都需要修改测试用例代码
  // expect(generateConfig()).toEqual({
  //   server: 'http://localhost',
  //   port: 8080,
  //   domain: 'localhost'
  // })
  // 2.使用快照匹配
  expect(generateAnotherConfig()).toMatchSnapshot();
});

import { generateAnotherConfig } from "../../src/JestBasic/snapts";

test("测试 generateAnotherConfig ", () => {
  expect(generateAnotherConfig()).toMatchInlineSnapshot(
    {
      time: expect.any(Date),
    },
    `
    Object {
      "age": 20,
      "domain": "localhost",
      "port": 8080,
      "server": "http://localhost",
      "time": Any<Date>,
    }
  `
  );
});
