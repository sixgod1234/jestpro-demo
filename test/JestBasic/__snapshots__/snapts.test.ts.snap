// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`测试 generateConfig  1`] = `
Object {
  "age": 20,
  "domain": "localhost",
  "port": 8080,
  "server": "http://localhost",
  "time": 2021-11-15T03:03:33.016Z,
}
`;
